tilemaps/walls.gd

```diff
@@ -106,14 +106,46 @@ func _ready():
 		set_cell(Vector2i(room.right_gate.y, room.right_gate.x), SOURCE_ID, decimal_to_cord[ROOM_RIGHT_GATE], 0)
 		set_cell(Vector2i(room.left_gate.y, room.left_gate.x), SOURCE_ID, decimal_to_cord[ROOM_LEFT_GATE], 0)
 
-	stair_position_in_world = self.map_to_local(stair_position_in_tilemap)
+	var stair_position_in_world = self.map_to_local(stair_position_in_tilemap)
 	emit_signal("stair_decided", stair_position_in_world)
+
+######
+# Functions used to ask questions about tiles in other nodes
+######
+'''# values set from binary mapping
+const NORTH_DIR = 4
+const SOUTH_DIR = 1
+const EAST_DIR = 2
+const WEST_DIR = 8'''
+# SENW
+# 1 means closed direction
+func directions_closed_at_pos(tilemap_pos: Vector2i) -> Array[int]:
+	var tile_data = self.get_cell_tile_data(tilemap_pos).get_custom_data('directions_closed')
```

Comment:
I need to know the directions closed to know which mineral I put. Basically if north is closed, i can put a mineral with the top texture to match it. I did it by manually typing **custom data** on the tilemap

![image](https://github.com/user-attachments/assets/17e5119c-4d18-4e66-94fa-7c774428594c)